### PR TITLE
Add terminal-first syu search command

### PR DESCRIPTION
**Pull request by tohboeh5**
## Summary\n- add a new `syu search` CLI for terminal-first lookup across IDs, titles, summaries, and descriptions\n- keep search behavior aligned with the browser search fields while supporting optional kind scoping and JSON output\n- document and trace the new command through the checked-in spec, generated docs, README, and tests\n\n## Validation\n- bash scripts/ci/quality-gates.sh\n- cargo run -- search source tests/fixtures/workspaces/passing --kind feature\n- cargo run -- search traceability . --kind requirement\n\nCloses #168

### CODE REVIEW

**Comment by tohboeh5** [comment]
Overview: This is a strong feature addition overall. The command fits the existing CLI vocabulary well, the kind scoping/JSON output make sense for automation, and the spec/doc updates keep the change easy to follow. I also confirmed the main happy paths locally. The weak spot is test completeness: the repository requires a 100% coverage gate, and this PR still leaves the philosophy-positive branch plus the whitespace-only error path uncovered, so CI stays red. Once those gaps are covered, this should be in much better shape.

File: src/command/lookup.rs [hunk 1]
```diff
@@ -214,6 +244,67 @@ impl<'a> WorkspaceLookup<'a> {
...
     }
+
+    fn extend_search_results(self, kind: LookupKind, query: &str, results: &mut Vec<SearchResult>) {
+        match kind {
+            LookupKind::Philosophy => {
+                for item in &self.workspace.philosophies {
+                    if field_matches_query(&item.id, query)
+                        || field_matches_query(&item.title, query)
+                    {
+                        results.push(SearchResult {
```
Comment (tohboeh5) [inline src/command/lookup.rs:255]: This new positive LookupKind::Philosophy branch is still uncovered. The current suite only proves that philosophy free-text fields stay out of scope, so cargo llvm-cov is still missing src/command/lookup.rs:254-259 and the coverage gate fails. Please add at least one positive philosophy ID/title search case so this branch is exercised and CI can go green.

File: src/command/search.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    query: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    kind: Option<&'static str>,
+    items: Vec<SearchResult>,
+}
+
+pub fn run_search_command(args: &SearchArgs) -> Result<i32> {
+    let query = args.query.trim();
+    if query.is_empty() {
+        bail!("search query must not be empty or whitespace");
```
Comment (tohboeh5) [inline src/command/search.rs:25]: This whitespace-only guard is untested right now. I reproduced the CLI path locally and it exits with the expected error, but without a regression test this branch remains uncovered at src/command/search.rs:25, which is enough to keep the pre-push and coverage jobs red under the repo 100 percent line-coverage gate. Please add a command test for the whitespace-only query path.

**Comment by tohboeh5** [comment]
Follow-up: GitHub blocked a formal review state here because this account is also the PR author. The two inline review comments above are the blocking items I found: one missing positive philosophy-search coverage case and one missing whitespace-query regression test.
